### ISSUE TITLE
Plugin with id 'com.gradle.build-scan' not found.

### ISSUE THREAD

**Issue by jaredsburrows** [gradle/gradle#3759]
## Repro:
`git clone https://github.com/jaredsburrows/android-gif-example`
`cd android-gif-example`
`gradlew assembleDebug --scan`

## Example build:
https://travis-ci.org/jaredsburrows/android-gif-example/builds/313230232?utm_source=github_status&utm_medium=notification

## Error:
```
$ gradlew assembleDebug --scan
Download https://plugins.gradle.org/m2/com/gradle/build-scan-plugin/1.10.3/build-scan-plugin-1.10.3.pom
Download https://plugins.gradle.org/m2/com/gradle/build-scan-plugin/1.10.3/build-scan-plugin-1.10.3.jar

FAILURE: Build failed with an exception.

* Where:
Auto-applied by using --scan

* What went wrong:
Could not apply requested plugin [id: 'com.gradle.build-scan', version: '1.10.3', artifact: 'com.gradle:build-scan-plugin:1.10.3'] as it does not provide a plugin with id 'com.gradle.build-scan'. This is caused by an incorrect plugin implementation. Please contact the plugin author(s).
> Plugin with id 'com.gradle.build-scan' not found.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
```

**Comment by oehme** [comment]
@bamboo It seems like the Kotlin DSL is trying to execute plugin requests for [second level scripts](https://travis-ci.org/jaredsburrows/android-gif-example/builds/313230232#L714). This should be avoided by [this commit](https://github.com/gradle/kotlin-dsl/commit/436a721487d649f412c3faf3b12b5c7c6b148171), did this not make it into 4.4?

**Comment by oehme** [comment]
@jaredsburrows You are reporting this for 4.4, but the build log says you are using 4.3.1. Can you please double-check whether it occurs in 4.4?

**Comment by bamboo** [comment]
@oehme I'm afraid that logic was lost during the `settings.gradle.kts` effort.

**Comment by oehme** [comment]
@bamboo What about the test case that was in that commit? Was that not making the right assertion? Or was it removed?

**Comment by bamboo** [comment]
It wasn't removed. That commit was made to a release branch and not backported to `develop` by mistake.

**Comment by oehme** [comment]
Ah I see, so that's why it is in 4.3.1, but not in 4.4. I think we'll need a 4.4.1 for this.

**Comment by bamboo** [comment]
We're working on a fix.

**Comment by bamboo** [comment]
The fix is ready. I'm cooking a new `kotlin-dsl` `0.13.2` release with the fix to go in `gradle@release`.

**Comment by jaredsburrows** [comment]
Awesome! So hopefully a Gradle 4.4.1 release soon?

**Comment by jaredsburrows** [comment]
I see that this is tagged with 4.5 RC1, is the fix too big for a patch on 4.4?

**Comment by bamboo** [comment]
@jaredsburrows It's not yet clear if there will be a 4.4.1 release, in the meantime the `v0.13.2` Kotlin DSL distribution snapshot is available: https://github.com/gradle/kotlin-dsl/releases/tag/v0.13.2

**Comment by jaredsburrows** [comment]
How can I use that with the latest Gradle release?

**Comment by bamboo** [comment]
You can't.

You can use the Gradle Kotlin DSL distribution snapshot mentioned in the release notes to validate the fix though:

    $ cd $YOUR_PROJECT_ROOT
    $ gradle wrapper --gradle-distribution-url https://repo.gradle.org/gradle/dist-snapshots/gradle-kotlin-dsl-4.4-20171208150335+0000-all.zip

It is basically Gradle 4.4 with Kotlin DSL v0.13.2.

**Comment by jaredsburrows** [comment]
@bamboo I have confirmed that this works. Thanks! I am now closing.

**Comment by oehme** [comment]
Reopening since this is not delivered.

**Comment by bamboo** [comment]
Resolved in 435ac55a9091551ef9ebba089b44007b082ed4ac.
